lib/utils/FileCache.mjs

```diff
@@ -59,24 +59,59 @@ export default class FileCache {
 	hasFileChanged(absoluteFilepath) {
 		// Get file descriptor compares current metadata against cached
 		// one and stores the result to "changed" prop.w
+
+		const fileCheckSumChanged = this._useCheckSum
+			? this.hasFileCheckSumChanged(absoluteFilepath)
+			: false;
 		const descriptor = this._fileCache.getFileDescriptor(absoluteFilepath);
-		/** @type {CacheMetadata} */
-		const meta = descriptor.meta || {};
-		const changed = descriptor.changed || meta.hashOfConfig !== this._hashOfConfig;
+
+		descriptor.meta.data ??= {};
+
+		assert(isPlainObject(descriptor.meta.data));
```

Comment:
[suggestion] How about making the following assertion unnecessary like this?

```suggestion
		/** @type {{ hashOfConfig?: string; }} */
		const metadata = descriptor.meta.data ??= {};
```

The following changes would be necessary too:

```diff
- descriptor.meta.data.hashOfConfig
+ metadata.hashOfConfig
```

Reply:
perfect! thank you!